build_resolvers/lib/src/resolver.dart

```diff
@@ -29,6 +30,12 @@ import 'analysis_driver.dart';
 import 'build_asset_uri_resolver.dart';
 import 'human_readable_duration.dart';
 
+// Used to protect all usages of the analysis driver from
+// `InconsistentAnalysisException` errors, which might occur if we are in the
+// middle of some calls to `changeFile` but have not yet completed
+// `applyPendingFileChanges`.
+final _driverPool = Pool(1);
```

Comment:
Could we have this as a field on `AnalyzerResolver`?

We don't have any use cases for it, but the rest of the code does (or should?) allow for multiple instances of the resolver to be independent of one another.

Reply:
Oh true, I think this could be an field on `AnalyzerResolver`, I can change that.

Reply:
done